script/gen_requirements_all.py

```diff
@@ -71,20 +76,55 @@ def main():
         print('\n'.join(errors))
         return
 
-    print('# Home Assistant core')
-    print('\n'.join(core_requirements()))
-    print()
-
+    output.append('# Home Assistant core')
+    output.append('\n')
+    output.append('\n'.join(core_requirements()))
+    output.append('\n')
     for pkg, requirements in reqs.items():
         for req in sorted(requirements,
                           key=lambda name: (len(name.split('.')), name)):
-            print('#', req)
+            output.append('\n# {}'.format(req))
 
         if comment_requirement(pkg):
-            print('#', pkg)
+            output.append('\n# {}\n'.format(pkg))
         else:
-            print(pkg)
-        print()
+            output.append('\n{}\n'.format(pkg))
+
+    return ''.join(output)
+
+
+def write_file(data):
+    """ Writes the modules to the requirements_all.txt. """
+    with open('requirements_all.txt', 'w+') as file:
```

Comment:
Files in the script directory will not be checked by `pylint` and `pep8`.

Do not overwrite a built-in with a variable (`file` in this case)